lessons/pydata/pandas/index.ipynb

```diff
@@ -0,0 +1,2872 @@
+       "    }\n",
+       "\n",
+       "    .dataframe tbody tr th {\n",
+       "        vertical-align: top;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>id</th>\n",
+       "      <th>name</th>\n",
+       "      <th>height</th>\n",
+       "      <th>weight</th>\n",
+       "      <th>color</th>\n",
+       "      <th>shape</th>\n",
+       "      <th>is baby</th>\n",
+       "      <th>type 1</th>\n",
+       "      <th>type 2</th>\n",
+       "      <th>hp</th>\n",
+       "      <th>attack</th>\n",
+       "      <th>defense</th>\n",
+       "      <th>speed</th>\n",
+       "      <th>color_uppercase</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <td>290</td>\n",
+       "      <td>291</td>\n",
+       "      <td>ninjask</td>\n",
+       "      <td>0.8</td>\n",
+       "      <td>12.0</td>\n",
+       "      <td>yellow</td>\n",
+       "      <td>bug-wings</td>\n",
+       "      <td>False</td>\n",
+       "      <td>Bug</td>\n",
+       "      <td>Flying</td>\n",
+       "      <td>61</td>\n",
+       "      <td>90</td>\n",
+       "      <td>45</td>\n",
+       "      <td>160</td>\n",
+       "      <td>YELLOW</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <td>794</td>\n",
+       "      <td>795</td>\n",
+       "      <td>pheromosa</td>\n",
+       "      <td>1.8</td>\n",
+       "      <td>25.0</td>\n",
+       "      <td>white</td>\n",
+       "      <td>humanoid</td>\n",
+       "      <td>False</td>\n",
+       "      <td>Bug</td>\n",
+       "      <td>Fighting</td>\n",
+       "      <td>71</td>\n",
+       "      <td>137</td>\n",
+       "      <td>37</td>\n",
+       "      <td>151</td>\n",
+       "      <td>WHITE</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <td>385</td>\n",
+       "      <td>386</td>\n",
+       "      <td>deoxys</td>\n",
+       "      <td>1.7</td>\n",
+       "      <td>60.8</td>\n",
+       "      <td>red</td>\n",
+       "      <td>humanoid</td>\n",
+       "      <td>False</td>\n",
+       "      <td>Psychic</td>\n",
+       "      <td>NaN</td>\n",
+       "      <td>50</td>\n",
+       "      <td>150</td>\n",
+       "      <td>50</td>\n",
+       "      <td>150</td>\n",
+       "      <td>RED</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <td>100</td>\n",
+       "      <td>101</td>\n",
+       "      <td>electrode</td>\n",
+       "      <td>1.2</td>\n",
+       "      <td>66.6</td>\n",
+       "      <td>red</td>\n",
+       "      <td>ball</td>\n",
+       "      <td>False</td>\n",
+       "      <td>Electric</td>\n",
+       "      <td>NaN</td>\n",
+       "      <td>60</td>\n"
```

Comment:
Podle mě je používání `inplace` zastaralé / nesprávné - v moderním Pandas vnímám jasnou tendenci používat "pipeline" přístup, tj. nemutovat. Mohlo by tam třeba být i `sorted_data = data.set_index("name").sort_index()`

Reply:
Nemyslím si, podle mě se nejde vyhnout ani jednomu přístupu. Pokud bych při práci s pandas nepoužíval inplace vůbec, trávil bych kopec času vymýšlením názvů pro proměnné, které potřebuji třeba jen pro jeden následující krok. O zbytečných kopiích DataFrame v paměti nemluvě.

Reply:
A opět, je to jen ukázka toho, co je možné, navíc je tam explicitně napsáno `… v tomto případě …`.

Reply:
Ukázka, co je možné, by ale raději neměla ukazovat, že jsou "možné" discouraged practices. Lze o tom diskutovat někde později, ale tady prostě chceš jen best practices. 

https://www.dataschool.io/future-of-pandas/#inplace

Reply:
Chlapi, pridávam svoj názor a to, že by som `inplace` *nepoužíval* tiež.

Moje argumenty by boli:
1. Sami autori pandas chcú `inplace` do budúcnosti odstrániť, pretože prináša zbytočnú komplexitu bez performance benefitov.
    viz: https://github.com/pandas-dev/pandas/issues/16529
2. Je dobrou praxou nemutovať. Je rozdiel, keď niekto skúsení ide proti dobrej praxi, pretože to väčšinou dobre zdokumentuje a má k tomu dôvod. Začiatočník ale nebude schopný rozlíšiť prečo nemá `inplace` používať a pre ktoré prípady platí `... v tomto prípade ...`.
3. Pre mňa je omnoho prijateľnejšie, že má začiatočník v pämati 15 rôznych (malých) DataFramov, ak si tým trénuje schopnosť pomenovávať premenné a systematicky si uvedomovať, v ktorej fáze preprocessingu sa mu ten DataFrame nachádza. Pri produkčných data pipelinoch majú tieto schopnosti cenu zlata.
4. Syntakticky podľa mňa vyzerá oveľa lepšie:
    `training_set = data.drop(target_variable, axis='columns')`
    než toto:
    `data.drop(target_variable, axis='columns', inplace=True)`

Keďze ale hlavne na internete a občas aj v publikáciach sa inplace vyskytne, tak by som zvážil, či by nebolo fajn to v tom kurze zmieniť, ale že je to považované za zlú prax, ale často na to narazíte...